### PR TITLE
Moonray docs 

### PR DESCRIPTION
**Pull request by evilnick**
- adds moonray sphinx instance for generating docs
- adds substitution terms
- converts existing docs to use substitution
- fixes a few CAPI docs

Currently the Moonray docs are available here - https://canonical-moonray.readthedocs-hosted.com/docs-moonray-staging/
N.B. there is minimal content there yet

NB this replaces #578

### CODE REVIEW

**Comment by evilnick** [comment]
Also adds a workflow to integrate standard doc test - I have left this as run on demand at the moment because it is going to spew out millions of errors

**Comment by evilnick** [comment]
You can check no Canonical kubernetes was hurt during this operation by viewing [these docs](https://canonical-ubuntu-documentation-library--581.com.readthedocs.build/canonical-kubernetes/581/capi/)
The new, largely empty Moonray stuff is [here](https://canonical-moonray--581.com.readthedocs.build/581/)

File: docs/moonray/explanation/about.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# What is {{product}}?
+
+At its core, {{product}} is a full implementation of upstream
+[Kubernetes] delivered in a compact, secure, reliable [snap] package. As the
```
Comment (louiseschmidtgen) [inline docs/moonray/explanation/about.md:4]: Could you please add the links for Kubernetes and the snap?

File: docs/moonray/howto/install.md [hunk 2]
```diff
@@ -0,0 +1,3 @@
+```{include} ../../src/snap/howto/install/snap.md
+
```
Comment (louiseschmidtgen) [inline docs/moonray/howto/install.md:2]: image
The note's Installing leads us to the how-to page, there's probably a page missing could you please add a dummy missing page.

Comment (louiseschmidtgen) [reply docs/moonray/howto/install.md:2]: Same goes for the other links on this page. Do we need to bring in the channel page and bootstrap ref guide?

File: docs/moonray/index.md [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+
+<!-- IMAGES -->
+
+[logo]: https://assets.ubuntu.com/v1/843c77b6-juju-at-a-glace.svg
+
+<!-- LINKS -->
+
+[Code of Conduct]: https://ubuntu.com/community/ethos/code-of-conduct
+[contribute]: index
```
Comment (louiseschmidtgen) [inline docs/moonray/index.md:74]: Let's add a TODO here or add the contribution page.

File: docs/moonray/index.md [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+[overview page]: /explanation/about
```
Comment (louiseschmidtgen) [inline docs/moonray/index.md:75]: This doesn't render. Let's add a TODO here or add the contribution page please.

**Review by louiseschmidtgen — COMMENTED** [review]
This is looking great, please add some TODOs or add missing files so that we don't accidentally forget about broken links.
Please check the the mirrored pages like `../../src/snap/howto/install/snap.md` and their links.

File: docs/moonray/howto/install.md (hunk 2 above)
Comment (evilnick) [reply docs/moonray/howto/install.md:2]: I think we will either need to put conditional blocks in there or harmonise the directory structure first. I have just added a todo note for now - bringing in those pages will bring in more unresolvable links right now

File: docs/moonray/index.md (hunk 4 above)
Comment (evilnick) [reply docs/moonray/index.md:75]: I'm just going to cut the bits we haven't added content for at this point

**Comment by evilnick** [comment]
thanks @louiseschmidtgen. I added a TODO on one file and just removed a lot of references elsewhere to resolve the broken links

Some of how this is going to work moving forward is going to require some careful editing of the 'common' pages and probably some changes to the structure so that things just work, but thats for another PR

**Review by bschimke95 — COMMENTED** [review]
great work @evilnick 

An initial question:

What is the reason for the moonray docs to be in `docs/moonray` and duplicate the setup like this? Looking at the code, it seems a bit odd to have `docs/src/...` and `docs/moonray`. I'd either move `moonray` to the `src` folder or rename `src` to something that reflects that these are the default docs (e.g. `core`?).

File: .github/workflows/automatic-doc-checks.yml [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  documentation-checks:
+    uses: canonical/documentation-workflows/.github/workflows/documentation-checks.yaml@main
+    with:
+      working-directory: 'docs/moonray'
```
[the remark this answers is not in the document]
Comment (evilnick) [reply .github/workflows/automatic-doc-checks.yml]: :+1:

**Comment by bschimke95** [comment]
Thanks for the context @evilnick 
The envisioned structure makes a lot of sense and also splitting up that work into multiple PRs is very reasonable.

**Review by bschimke95 — APPROVED** [review]
LGTM

**Pull request merged**